spec/views/post_index_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :feature do
+  before(:each) do
+    @user1 = User.create(
+      name: 'Promise',
+      bio: 'Fullstack Developer',
+      photo: 'http://promise.com',
+      posts_counter: 2
+    )
+
+    @post1 = Post.create(
+      author: @user1,
+      title: 'First post',
+      text: 'This is my first post',
+      comments_counter: 0,
+      likes_counter: 0
+    )
+
+    @post2 = Post.create(
+      author: @user1,
+      title: 'Second post',
+      text: 'This is my second post',
+      comments_counter: 0,
+      likes_counter: 0
+    )
+
+    Comment.create(post: @post1, author: @user1, text: 'Hi everytone, nice to meet you!')
+    Comment.create(post: @post1, author: @user1, text: 'Hello loves, welcome!!')
+    Comment.create(post: @post2, author: @user1, text: 'How are you?, been a while')
+
+    Like.create(post: @post1, author: @user1)
+    Like.create(post: @post2, author: @user1)
+  end
+
+  describe 'a post index page' do
+    it 'displays the user photo' do
+      visit user_posts_path(@user1)
+      expect(page).to have_css("img[src*='http://promise.com']")
+    end
+
+    it 'displays the user name' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content('Promise')
+    end
+
+    it 'displays total posts by the user has written' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content('2')
+    end
+
```

Comment:
- [ ] great job. kindly add  Tests that asserts you can see a post's title

Reply:
Implemented this !